doc/source/ScriptingAPI/index.md

```diff
@@ -0,0 +1,51 @@
+# Scripting API #
+
+Gaffer's framework is built from a combination of C++ and Python. A significant portion of the core architecture has hooks in Python: much of the GUI is written in Python, nodes are wrapped in Python, and the Gaffer file formats are simple Python scripts.
+
+There is a direct one-to-one correspondence between Gaffer's C++ and Python APIs, so if you start out using one language, you can easily transfer your work to the other. It is relatively straightforward to prototype in Python, and later convert to C++ if performance becomes an issue. Or, alternatively, you can spend most of your time coding in C++, and switch to Python when greater ease and flexibility are needed.
```

Comment:
I think I'm going to need an opinion from @johnhaddon on this comment.

It's probably important to mention that we highly discourage implementing any nodes that actually compute plug values in Python - John used to actually disallow this completely, but I think it is now possible, and is sometimes done in Jabuka?  But even there, we try to use Python expressions instead of nodes implemented completely in Python ( the Python expression node has a very conservative hash function implemented in C++, which is often dramatically faster than even a really smart hash function written in Python ).

But without getting into that complexity, I think the message that we should be sending is that you can do a huge amount with Gaffer by assembling node networks and scripting ( and even implementing your own Python nodes that are implemented as internal networks of built in Gaffer nodes ).  But if you actually need to implement a new kind of compute node, then we do almost require you to use C++.